rubyforgood/diaper
# Add Item Visibility

Resolves #1743 

This work was done by both me and @egiurleo 

### Description
1. Added a checkbox in the UI to allow admins to mark an inventory item as visible or invisible to partners.
2. Updated partner request api to not show the item that has been marked invisible

### Type of change

* New feature (non-breaking change which adds functionality)


### How Has This Been Tested?
We have added the following tests:
* System spec to uncheck the `Visible to partners` checkbox
* Controller spec to update the `visible_to_partners` field to true and false
* Requests spec to check partner request API only returns visible items

We have conducted a manual integration test locally:
* Spin up diaper and partner app locally using `puma-dev`
* Mark one of the inventory items as invisible to partner
* Verify in the partner app that the items are not visible in the items dropdown

### Screenshots
![2020-09-06_12-32-15 (1)](https://user-images.githubusercontent.com/1865469/92331584-5a7c7a80-f03d-11ea-8f1a-0ec37f05264b.gif)
